### PR TITLE
Fix tf shared embedding

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
A hack was used to properly import the shared embedding weights, but it can be removed (removing it also is convenient for the sharding PR) 

Found this while testing #17713. In HF's `save_pretrained` and  `load_pretrained` the layer name is changed using ` name = "/".join(weight_name.split("/")[1:])`. This was breaking for OPT as the layer name was ` 'decoder.embed_tokens/model.decoder.embed_tokens/weight:0'` instead of `'tfopt_model/model/decoder/embed_tokens/weight:0'`. The naming is strange and had to use a scope hack. The hack comes from `BART`

### CODE REVIEW

**Comment by ArthurZucker** [comment]
All the TF weights of OPT will need to be updated if this is approved. I think I can handle that along with #17713.

File: tests/models/opt/test_modeling_tf_opt.py [hunk 1]
```diff
@@ -281,14 +281,14 @@ def test_inference_no_head(self):
 
 
 @require_tf
-@slow
+# @slow
```
Comment (sgugger) [inline tests/models/opt/test_modeling_tf_opt.py]: Will need to be uncommented before merging.

File: tests/models/opt/test_modeling_tf_opt.py [hunk 2]
```diff
@@ -281,14 +281,14 @@ def test_inference_no_head(self):
...
 class TFOPTEmbeddingsTest(unittest.TestCase):
     def setUp(self):
         super().setUp()
         self.path_model = "facebook/opt-350m"
 
     def test_logits(self):
-        model = TFOPTForCausalLM.from_pretrained(self.path_model)
+        model = TFOPTForCausalLM.from_pretrained(self.path_model, from_pt=True)
```
Comment (sgugger) [inline tests/models/opt/test_modeling_tf_opt.py]: Those will need to disappear after the model weights re-upload, but should probably be in a separate PR to merge this one quickly.

**Review by sgugger — APPROVED** [review]
Thanks for fixing!

File: tests/models/opt/test_modeling_tf_opt.py (hunk 1 above)
Comment (patrickvonplaten) [reply tests/models/opt/test_modeling_tf_opt.py]: Think it's already fixed in the new commit

File: tests/models/opt/test_modeling_tf_opt.py (hunk 2 above)
Comment (patrickvonplaten) [reply tests/models/opt/test_modeling_tf_opt.py]: should also be fixed in the new commit

**Review by patrickvonplaten — APPROVED** [review]
